src/components/views/rooms/EventTile.js

```diff
@@ -742,7 +742,11 @@ function E2ePadlockUnencrypted(props) {
 }
 
 function E2ePadlock(props) {
-    return <img className="mx_EventTile_e2eIcon" {...props} />;
+    if (SettingsStore.getValue("alwaysShowEncryptionIcons")) {
+        return <img className="mx_EventTile_e2eIcon" {...props} />;
+    } else {
+        return <img className="mx_EventTile_e2eIcon" hidden {...props} />;
```

Comment:
Having CSS that is conditional on the properties is coupling that should be avoided. If the location for the images changes or someone removes `hidden`, thinking it's only needed in the JS, the CSS might go unchecked and regressions could appear.

Please use `className`.

Reply:
Done.

Of note:

I think the `hidden` attribute is a better approach in general, but it suffers from some weaknesses due to how the CSS in this codebase is structured. In general, the attribute is better because this scenario is exactly what it is designed for. Replicating a native functionality with a class is not generally a very good thing to do.

What trips us up here is the specificity of the other CSS in the codebase. The original commit was forced to explicitly set the `hidden` attribute to `display: none`, which is not normally necessary for `hidden` to work. Normally other CSS isn't overriding the `hidden` attribute in this way, but due to the way other CSS in the codebase was structured, it was necessary here.

Had there not been other stuff interfering with the native functionality of `hidden`, the entire rule setting it to `display: none` would not be necessary.